src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -20,20 +20,31 @@ internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedO
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioExperimentationService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
-            : base(threadingContext, assertIsForeground: true)
+            : base(threadingContext)
         {
-            try
+            object experimentationServiceOpt = null;
+            MethodInfo isCachedFlightEnabledInfo = null;
+
+            threadingContext.JoinableTaskFactory.Run(async () =>
             {
-                _experimentationServiceOpt = serviceProvider.GetService(typeof(SVsExperimentationService));
-                if (_experimentationServiceOpt != null)
+                await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                try
                 {
-                    _isCachedFlightEnabledInfo = _experimentationServiceOpt.GetType().GetMethod(
-                        "IsCachedFlightEnabled", BindingFlags.Public | BindingFlags.Instance);
+                    experimentationServiceOpt = serviceProvider.GetService(typeof(SVsExperimentationService));
+                    if (experimentationServiceOpt != null)
+                    {
+                        isCachedFlightEnabledInfo = experimentationServiceOpt.GetType().GetMethod(
+                            "IsCachedFlightEnabled", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
aside: is reflection necessary anymore?  It feels pretty weird that we would need this isntead of just calling the real interface now.

Reply:
➡️ I wasn't able to find an appropriate definition of `IVsExperimentationService` to use.